src/snabbdom.ts

```diff
@@ -57,10 +57,29 @@ export function init(modules: Array<Partial<Module>>, domApi?: DOMAPI) {
     }
   }
 
-  function emptyNodeAt(elm: Element) {
+  function toVNode(elm: Element): VNode {
     const id = elm.id ? '#' + elm.id : '';
-    const c = elm.className ? '.' + elm.className.split(' ').join('.') : '';
-    return vnode(api.tagName(elm).toLowerCase() + id + c, {}, [], undefined, elm);
+    const cn = elm.getAttribute('class');
+    const c = cn ? '.' + cn.split(' ').join('.') : '';
+    const sel = api.tagName(elm).toLowerCase() + id + c;
+    const attrs: any = {};
+    for (let i = 0, elmAttrs = elm.attributes, n = elmAttrs.length; i < n; i++) {
+      attrs[elmAttrs[i].nodeName] = elmAttrs[i].nodeValue;
+    }
+    delete attrs.id;
```

Comment:
`delete` might have undesirable "perf" consequences (lead to deoptimization) in dictionary-like objects like `attrs`. Instead of adding the keys and then delete them immediately after, just opt-out of the addition of the key in the first place (maybe an `if` statement in line 67). Btw, I'm not sure this is a critical path for the engine where the `delete` will have an impact.